### ISSUE TITLE
JetBrains like Typescript file organization 

### ISSUE THREAD

**Issue by vezinadj** [Microsoft/vscode#5093]
Not sure if this is more of a plugin request or if this could be a feature right into Code. I have been bouncing between RubyMine and VisualStudioCode (Almost 100% in VisualStudioCode now) and I will say, the file organization in RubyMine for compiled Typescript .ts files is awesome.

It would be a great feature if you had the option to have a .ts file fold down. In RubyMine, after a typescript compile occurs and you don't have a specific output path set in your tsconfig.json, the .ts contains a dropdown arrow, where the compiled .js.map and .js files are underneath the .ts file.

This cleans up the directory tree and makes it much easier to navigate files.

Basically, if a .ts file is present in the folder, and a .js and .js.map file is present with the same base file name, you could put them underneath the .ts file.

**Comment by DickvdBrink** [comment]
Not the same as your suggestion but this might be enough for you (at least it was for me ;))!
You can create a workspace settings file and add the following to it:
```
"files.exclude": {
	"**/*.js": { "when": "$(basename).ts"}
}
```
This way the js files aren't cluttering the treeview when there is a corresponding `.ts` file. Only helps if you don't care about the `.js` file because you can't really see it anymore.

Is that enough or do you really want the dropdown/folding thingy?

**Comment by vezinadj** [comment]
Thank you for the suggestion, I think its a good band aid for right now. Having the option to see the javascript files would still be great. 

I find myself checking over the compiled javascript files every once in awhile so it would be great to have the ability to see them. Just in a more organized fashion.

**Comment by bpasero** [comment]
#4227
